### DISCUSSION TITLE
pi pico with waveshare pico-lcd-2 display problem

### DISCUSSION THREAD

**Discussion by cuu123** [Bodmer/TFT_eSPI#2483] [Raspberry Pi Pico]
pi pico with waveshare pico-lcd-2 display is not working. howto correct? please help
here is the setup file
my test file is examples\Test and diagnostics\Colour_Test

```
#define USER_SETUP_INFO "User_Setup"

// Display type -  only define if RPi display
#define RPI_DISPLAY_TYPE // 20MHz maximum SPI

#define ST7789_2_DRIVER    // Minimal configuration option, define additional parameters below for this display

#define TFT_WIDTH  240 // ST7789 240 x 240 and 240 x 320
#define TFT_HEIGHT 320 // ST7789 240 x 320

// https://www.waveshare.net/shop/Pico-LCD-2.htm
#define TFT_DC    8  // Data Command control pin·
#define TFT_CS    9  // Chip select control pin
#define TFT_SCLK 10
#define TFT_MOSI 11
//#define TFT_MISO 12
#define TFT_RST  12 // Reset pin (could connect to RST pin)
#define TFT_BL   13  // LED back-light

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:-.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
//#define LOAD_FONT8N // Font 8. Alternative to Font 8 above, slightly narrower, so 3 digits fit a 160 pixel TFT
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

// Comment out the #define below to stop the SPIFFS filing system and smooth font code being loaded
// this will save ~20kbytes of FLASH
#define SMOOTH_FONT

// For the RP2040 processor define the SPI port channel used (default 0 if undefined)
#define TFT_SPI_PORT 1 // Set to 0 if SPI0 pins are used, or 1 if spi1 pins used

#define SPI_FREQUENCY       20000000
#define SPI_READ_FREQUENCY  20000000
#define SPI_TOUCH_FREQUENCY  2500000
```
![图片](https://user-images.githubusercontent.com/40519958/223880569-4e50cf55-b95e-4535-ac6a-2b4d4152b77b.png)

**Comment by Bodmer** [comment]
Have you followed the instructions here:
https://www.waveshare.com/wiki/Pico-ResTouch-LCD-3.5

Have you run the Read_User_Setup sketch and checked the serial output is correct?

**Comment by cuu123** [comment]
Thanks for your reply. I followed the instructions here:
https://www.waveshare.com/wiki/Pico-LCD-2

And I run the Read_User_Setup sketch and paste the serial output as follows:
Please help me find out what the problem is.

[code]
TFT_eSPI ver = 2.5.23
Processor    = RP2040
Transactions = Yes
Interface    = SPI
Display driver = 778B
Display width  = 240
Display height = 320

MOSI    = GPIO 11
SCK     = GPIO 10
TFT_CS   = GPIO 9
TFT_DC   = GPIO 8
TFT_RST  = GPIO 12

TFT_BL           = GPIO 13

Font GLCD   loaded
Font 2      loaded
Font 4      loaded
Font 6      loaded
Font 7      loaded
Font 8      loaded
Smooth font enabled

Display SPI frequency = 20.00
[/code]

**Comment by Bodmer** [comment]
Are you using the Arduino IDE and Earle Philhower's board package?

**Comment by cuu123** [comment]
Yes, Arduino IDE 1.8.19 and Earle Philhower's board package 3.0.0.

I confirmed the environment and rerun the Read_User_Setup sketch.  

[code]
TFT_eSPI ver = 2.5.23
Processor    = RP2040
Transactions = Yes
Interface    = SPI
Display driver = 778B
Display width  = 240
Display height = 320

MOSI    = GPIO 11
SCK     = GPIO 10
TFT_CS   = GPIO 9
TFT_DC   = GPIO 8
TFT_RST  = GPIO 12

TFT_BL           = GPIO 13

Font GLCD   loaded
Font 2      loaded
Font 4      loaded
Font 6      loaded
Font 7      loaded
Font 8      loaded
Smooth font enabled

Display SPI frequency = 20.00
[/code]

**Comment by Bodmer** [comment]
Comment out the following line in the setup file thus:
//#define RPI_DISPLAY_TYPE

**Comment by cuu123** [comment]
It works! Thanks for your kind help!!

![图片](https://user-images.githubusercontent.com/40519958/224318503-7ef3631e-057c-480d-ae00-f357f68c3c38.png)
![图片](https://user-images.githubusercontent.com/40519958/224318568-2720b935-8c30-4d64-82dd-69d1811e2a03.png)

**Comment by tzechienchu** [comment]
I use your User_setup.h, but still not working.
I change from use SPI Port to PIO_SPI 

//#define TFT_SPI_PORT 1
#define RP2040_PIO_SPI 

It work.
